dapp_runner/runner/service.py

```diff
@@ -79,26 +81,43 @@ async def _wait_for_termination(self):
             self._report_state_change()
 
 
+class HttpProxyDappService(DappService, HttpProxyService):
+    """Yapapi Service definition enabling HTTP proxy for dapps."""
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+
 async def get_service(
     name: str, desc: ServiceDescriptor, payloads: Dict[str, Payload]
-) -> Type[DappService]:
+) -> Tuple[Type[Service], dict]:
     """Create a service class corresponding with its descriptor."""
 
     try:
         payload_instance = payloads[desc.payload]
     except KeyError:
         raise RunnerError(f"Undefined payload: `{desc.payload}`")
 
-    async def get_payload():
-        return payload_instance
+    service_instance_params: dict = {}
+    service_instance_params["entrypoint"] = desc.entrypoint
 
     DappServiceClass = type(
         f"DappService-{name}",
         (DappService,),
-        {
-            "get_payload": staticmethod(get_payload),
-            "entrypoint": desc.entrypoint,
-        },
+        {},
     )
 
-    return DappServiceClass
+    if desc.http_proxy:
+        port_mapping = desc.http_proxy.ports[0]
```

Comment:
maybe we should throw an error for now saying that multiple port mappings are currently unsupported?

Reply:
Good idea, will do!

Reply:
Added in c215bd1